workflows/src/main/kotlin/com/flows/BuildRoadFlow.kt

```diff
@@ -40,6 +42,11 @@ class BuildRoadFlow(val gameBoardLinearId: UniqueIdentifier, val hexTileIndex: I
         val gameBoardReferenceStateAndRef = ReferencedStateAndRef(gameBoardStateAndRef)
         val gameBoardState = gameBoardStateAndRef.state.data
 
+        // Step *. Retrieve roads, settlements and current longest road state
+        val roadStates = serviceHub.vaultService.queryBy<RoadState>().states.map { it.state.data }
+        val settlementStates = serviceHub.vaultService.queryBy<SettlementState>().states.map { it.state.data }
```

Comment:
@rogersanick It makes me think that we should add a `gameBoardId` to our `SettlementState` and `RoadState` so that a state cannot be linked to more than 1 game. Otherwise, someone could have the same settlement in 2 games and go with whichever brings the more win.

Reply:
+1

Reply:
+100000 - I will add this to master once I have merged in @xavierlepretre latest PR

Reply:
Or perhaps you merge this one first, and I fix the conflicts before you merge mine.